app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala

```diff
@@ -72,18 +71,16 @@ class EoriNumberController @Inject()(
     (for {
       validated <- mibConnector.checkEoriNumber(eori)
       result    <- validateEoriAndRedirect(eori, isAgent, declarationType, validated)
-    } yield result).recover { case _ => badRequestResult(isAgent, declarationType) }
+    } yield result).recover { case _ => badRequestResult(isAgent, declarationType, eori) }
 
   private def validateEoriAndRedirect(eori: String, isAgent: YesNo, declarationType: DeclarationType, response: CheckResponse)(
     implicit request: DeclarationJourneyRequest[AnyContent]): Future[Result] =
     if (response.valid)
       persistAndRedirect(request.declarationJourney.copy(maybeEori = Some(Eori(eori))), routes.TravellerDetailsController.onPageLoad())
     else
-      Future.successful(badRequestResult(isAgent, declarationType))
+      Future.successful(badRequestResult(isAgent, declarationType, eori))
 
-  private def badRequestResult(isAgent: YesNo, declarationType: DeclarationType)(
-    implicit request: DeclarationJourneyRequest[AnyContent]): Result = {
-    val formWithError = form(isAgent, request.declarationType).withError(FormError("eori", "eoriNumber.error.notFound"))
-    BadRequest(view(formWithError, isAgent, backButtonUrl, declarationType))
-  }
+  private def badRequestResult(isAgent: YesNo, declarationType: DeclarationType, eori: String)(
+    implicit request: DeclarationJourneyRequest[AnyContent]): Result =
+    BadRequest(view(formWithError(isAgent, request.declarationType, eori), isAgent, backButtonUrl, declarationType))
```

Comment:
I think you've gone the long way around about this, does formWithError not have a .fill() method

so would just be BadRequest(view(formWithError(isAgent, request.declarationType).fill(eori)...

Reply:
or actually even better, when you declare your val formWithError just stick a .fill() after the .withError()

Reply:
you mean 
`form(customsAgent, declarationType)
      .withError(FormError(fieldName, "eoriNumber.error.notFound"))
      .bind(Map(fieldName -> eori))`
Vs
`form(customsAgent, declarationType)
      .withError(FormError(fieldName, "eoriNumber.error.notFound"))
      .fill(eori)`
???

Reply:
If I understand correctly, what you are trying to do is prepopulate the field with their EORI on the page

So you don't want to call bind, that will bind the form and return a String?

You instead want an instance of the form, with an error, but the form is filled with their previous submission